src/main/java/unicash/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,160 @@
+package unicash.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static unicash.logic.UniCashMessages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static unicash.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static unicash.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static unicash.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static unicash.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static unicash.logic.parser.CliSyntax.PREFIX_NAME;
+import static unicash.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.List;
+
+import unicash.commons.util.ToStringBuilder;
+import unicash.logic.commands.FilterCommand;
+import unicash.logic.parser.exceptions.ParseException;
+import unicash.model.category.Category;
+import unicash.model.commons.Amount;
+import unicash.model.transaction.DateTime;
+import unicash.model.transaction.Location;
+import unicash.model.transaction.Name;
+import unicash.model.transaction.Type;
+import unicash.model.transaction.predicates.TransactionContainsAllKeywordsPredicate;
+
+
+/**
+ * Parses input arguments and creates a new FilterCommand object
+ */
+public class FilterCommandParser implements Parser<FilterCommand> {
+
+    private TransactionContainsAllKeywordsPredicate filterPredicate;
+
+    /**
+     * Creates a {@code FilterCommandParser} object with a default
+     * {@code TransactionContainsAllKeywordsPredicate} predicate.
+     */
+    FilterCommandParser() {
+        this.filterPredicate = new TransactionContainsAllKeywordsPredicate();
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterCommand
+     * and returns a FilterCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_TYPE, PREFIX_AMOUNT, PREFIX_DATETIME,
+                        PREFIX_CATEGORY, PREFIX_LOCATION);
+
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty() || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterCommand.MESSAGE_USAGE));
+        }
+
+        // TODO: Propose an future implementation with more than 2 types of transactions in DG
```

Comment:
Hmm, not sure if this is needed since tP is coming to a close soon right? Maybe good to include in the docs for @lipwei1808 who made the `AddTransactionCommand` to add your rationale for not hardcoding the type of transaction

Reply:
Sorry, the TODO was supposed to be for writing the documentation appendix, to explain "why certain features were done certain way" and this would be along the process of explaining why FilterCommand is limited to one Type prefix only even though that makes it less narrowable (bc rn there are 2 mutually exclusive types, ANDing them would return 0 transactions, but if more types are introduced (i.e. transfers, loans, etc) in AddTransaction, this could change)